spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java

```diff
@@ -155,4 +162,45 @@ private void registerSqlExtensions(SparkConf conf) {
       conf.set(StaticSQLConf.SPARK_SESSION_EXTENSIONS().key(), extensionString);
     }
   }
+
+  private static GravitinoClient createGravitinoClient(
+      String uri, String metalake, SparkConf sparkConf) {
+    ClientBuilder builder = GravitinoClient.builder(uri).withMetalake(metalake);
+    String authType =
+        sparkConf.get(GravitinoSparkConfig.GRAVITINO_AUTH_TYPE, AuthProperties.SIMPLE_AUTH_TYPE);
+    if (AuthProperties.isSimple(authType)) {
+      String username = sparkConf.get(GravitinoSparkConfig.GRAVITINO_SIMPLE_USER_NAME, null);
+      if (StringUtils.isNotBlank(username)) {
+        builder.withSimpleAuth(username.trim());
+      } else {
+        builder.withSimpleAuth();
+      }
+    } else if (AuthProperties.isOAuth2(authType)) {
+      String oAuthUri = sparkConf.get(GravitinoSparkConfig.GRAVITINO_OAUTH2_URI, null);
+      String credential = sparkConf.get(GravitinoSparkConfig.GRAVITINO_OAUTH2_CREDENTIAL, null);
+      String path = sparkConf.get(GravitinoSparkConfig.GRAVITINO_OAUTH2_PATH, null);
+      String scope = sparkConf.get(GravitinoSparkConfig.GRAVITINO_OAUTH2_SCOPE, null);
+      DefaultOAuth2TokenProvider oAuth2TokenProvider =
+          DefaultOAuth2TokenProvider.builder()
+              .withUri(oAuthUri)
+              .withCredential(credential)
+              .withPath(path)
+              .withScope(scope)
+              .build();
+      builder.withOAuth(oAuth2TokenProvider);
+    } else if (AuthProperties.isKerberos(authType)) {
+      String principal = sparkConf.get(GravitinoSparkConfig.GRAVITINO_KERBEROS_PRINCIPAL, null);
```

Comment:
The default value `null` is unnecessary since null is NOT a valid value for `principal`, could you check the validity before creating a Gravitino client?

Reply:
done